parquet-column/src/main/java/org/apache/parquet/internal/column/columnindex/IndexIterator.java

```diff
@@ -55,6 +55,105 @@ static PrimitiveIterator.OfInt rangeTranslate(int from, int to, IntUnaryOperator
     return new IndexIterator(from, to + 1, i -> true, translator);
   }
 
+  static PrimitiveIterator.OfInt intersection(PrimitiveIterator.OfInt lhs, PrimitiveIterator.OfInt rhs) {
+    return new PrimitiveIterator.OfInt() {
+      private int next = fetchNext();
+
+      @Override
+      public int nextInt() {
+        int result = next;
```

Comment:
Sorry for bothering with the iterator contract again. It should throw `NoSuchElementException` if there are no more elements (instead of returning `-1`).

Reply:
thanks for catching that! Updated both.